### ISSUE TITLE
Self with a nested class is not recognized

### ISSUE THREAD

**Issue by AndreuCodina** [astral-sh/ty#2643]
### Summary

I use this when I have to define a class per test.

```python
from typing import Self


class TestPerson:
    def test_person(self) -> None:
        class Person:
            def get(self) -> Self:
                return self
```

Error:

```
error[invalid-return-type]: Return type does not match returned value
  |
5 |     def test_person(self) -> None:
6 |         class Person:
7 |             def get(self) -> Self:
  |                              ---- Expected `Self@test_person` because of return type
8 |                 return self
  |                        ^^^^ expected `Self@test_person`, found `Self@get`
  |
info: rule `invalid-return-type` is enabled by default
```

Playground: https://play.ty.dev/64f90de3-6239-47f8-97e5-5be3c64cdc95

### Version

0.0.14
